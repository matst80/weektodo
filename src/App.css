@import url('https://fonts.googleapis.com/css?family=Fjalla+One');
body {
  background: linear-gradient(6deg, #4b6cb7 0%, #182848 100%);
  font-family: 'Fjalla One', sans-serif;
  color: #fff;
}
input:focus{
  outline: 0;
}
.days, .center {
  display: flex;
  flex-direction: row;
  flex:1;
  height: 100vh;
}
.loading {
  flex:1;
  align-self: center;
  text-align: center;
}
.weeknr {
  flex:2;
}
.weekbutton {
  flex:1;
  text-align: center;
  cursor: pointer;
}
.center {
  padding: 10px;
  display: flex;
}
.currentweek {
  text-align: center;
  padding: 10px;
  display: flex;
  flex-direction: row;
}
.button {
  border-radius: 20px;
  padding: 10px;
  background-color: cadetblue;
}