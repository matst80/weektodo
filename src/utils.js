export const getWeek = (date) => {
    var newYear = new Date(date.getFullYear(), 0, 1);
    var day = newYear.getDay();
    day = (day >= 0 ? day : day + 7);
    var daynum = Math.floor((date.getTime() - newYear.getTime() -
        (date.getTimezoneOffset() - newYear.getTimezoneOffset()) * 60000) / 86400000) + 1;

    let weeknum = Math.floor((daynum + day - 1) / 7);
    if (day < 4) {
        weeknum = Math.floor((daynum + day - 1) / 7) + 1;
        if (weeknum > 52) {
            let nYear = new Date(date.getFullYear() + 1, 0, 1);
            let nday = nYear.getDay();
            nday = nday >= 0 ? nday : nday + 7;
            weeknum = nday < 4 ? 1 : 53;
        }
    }
    return weeknum;
};

export const dayNames = [
    'Måndag',
    'Tisdag',
    'Onsdag',
    'Torsdag',
    'Fredag',
    'Lördag',
    'Söndag'
];

export const forEach = function (obj, callback) {
    var ret = [];
    for (var key in obj) {
        var item = obj[key];
        var data = callback(item, key);
        if (data)
            ret.push(data);
    }
    return ret;
}

export const safeName = (val) =>
    val
        .toLowerCase()
        .replace(/å/ig, 'a')
        .replace(/ö/ig, 'o')
        .replace(/ä/ig, 'a');

const fakeUser = { fake: true };
const fakeAuth = {
    onAuthStateChanged: (cb) => { fakeAuth.cb = cb; },
    signInAnonymously: () => { fakeAuth.cb(fakeUser); return Promise.resolve(fakeUser); }
}
const fakeDb = {
    ref: _ => {
        return {
            on: (val, cb) => {
                cb({ exists: () => { return false; } });
            },
            off: () => { }
        }
    }
}

export const auth = window.firebase ? window.firebase.auth() : fakeAuth;
export const db = window.firebase ? window.firebase.database() : fakeDb;
