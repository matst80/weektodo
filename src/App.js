import React, { Component } from 'react';
import Day from './Day.js'
import { safeName, auth, dayNames, getWeek } from './utils';
import './App.css';

const safeNames = dayNames.map(safeName);

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      user: false,
      zoomed: false,
      week: props.week || getWeek(new Date())
    }
  }
  componentDidMount() {
    auth.onAuthStateChanged((user) => {
      this.setState({ user: user || false });
    });
    auth.signInAnonymously()
  }
  changeWeek = (week) => {
    this.setState({ week });
  }
  setZoom = (idx) => {
    let { zoomed } = this.state;
    zoomed = (zoomed === idx) ? false : idx;
    this.setState({ zoomed });
  }
  render() {
    const { user, week, zoomed } = this.state;
    if (!user) {
      return (
        <div className="center">
          <div className="loading">Loading...</div>
        </div>);
    }

    let days = dayNames.map((name, i) => (
      <Day
        onZoomed={_ => this.setZoom(i)}
        zoomed={{ isModal: zoomed === i, other: zoomed }}
        key={`${week}/${i}`}
        dbKey={`todo/${week}/${safeNames[i]}`}
        dayName={name}
      />));

    return (
      <div className="app">
        <div className="days">
          {days}
        </div>
        <div className="currentweek">
          {week > 0 && <div className="weekbutton" onClick={() => this.changeWeek(week - 1)}>&lsaquo;</div>}
          <div className="weeknr">Vecka {week}</div>
          {week < 52 && <div className="weekbutton" onClick={() => this.changeWeek(week + 1)}>&rsaquo;</div>}</div>
      </div>
    );
  }
}
