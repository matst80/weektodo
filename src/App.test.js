import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import './utils';
import renderer from 'react-test-renderer'
import { safeName, getWeek } from './utils';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly lastweek', () => {
  const tree = renderer
    .create(<App week={52} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly firstweek', () => {
  const tree = renderer
    .create(<App week={1} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

test('weekCalc', () => {
  expect(getWeek(new Date('2018/10/17'))).toBe(42);
});

test('no swedish chars in safeName', () => {
  expect(safeName('åäöÅÄÖ')).toBe('aaoaao');
});