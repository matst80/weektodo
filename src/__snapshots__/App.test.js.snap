// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="app"
>
  <div
    className="days"
  >
    <div
      className="day"
    >
      <div
        className="dayname"
        onClick={[Function]}
      >
        Måndag
      </div>
      <div
        className="todo"
      >
        <div
          className="list"
        />
        <input
          onChange={[Function]}
          onKeyPressCapture={[Function]}
          value=""
        />
      </div>
    </div>
    <div
      className="day"
    >
      <div
        className="dayname"
        onClick={[Function]}
      >
        Tisdag
      </div>
      <div
        className="todo"
      >
        <div
          className="list"
        />
        <input
          onChange={[Function]}
          onKeyPressCapture={[Function]}
          value=""
        />
      </div>
    </div>
    <div
      className="day"
    >
      <div
        className="dayname"
        onClick={[Function]}
      >
        Onsdag
      </div>
      <div
        className="todo"
      >
        <div
          className="list"
        />
        <input
          onChange={[Function]}
          onKeyPressCapture={[Function]}
          value=""
        />
      </div>
    </div>
    <div
      className="day"
    >
      <div
        className="dayname"
        onClick={[Function]}
      >
        Torsdag
      </div>
      <div
        className="todo"
      >
        <div
          className="list"
        />
        <input
          onChange={[Function]}
          onKeyPressCapture={[Function]}
          value=""
        />
      </div>
    </div>
    <div
      className="day"
    >
      <div
        className="dayname"
        onClick={[Function]}
      >
        Fredag
      </div>
      <div
        className="todo"
      >
        <div
          className="list"
        />
        <input
          onChange={[Function]}
          onKeyPressCapture={[Function]}
          value=""
        />
      </div>
    </div>
    <div
      className="day"
    >
      <div
        className="dayname"
        onClick={[Function]}
      >
        Lördag
      </div>
      <div
        className="todo"
      >
        <div
          className="list"
        />
        <input
          onChange={[Function]}
          onKeyPressCapture={[Function]}
          value=""
        />
      </div>
    </div>
    <div
      className="day"
    >
      <div
        className="dayname"
        onClick={[Function]}
      >
        Söndag
      </div>
      <div
        className="todo"
      >
        <div
          className="list"
        />
        <input
          onChange={[Function]}
          onKeyPressCapture={[Function]}
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="currentweek"
  >
    <div
      className="weekbutton"
      onClick={[Function]}
    >
      ‹
    </div>
    <div
      className="weeknr"
    >
      Vecka 
      42
    </div>
    <div
      className="weekbutton"
      onClick={[Function]}
    >
      ›
    </div>
  </div>
</div>
`;

exports[`renders correctly firstweek 1`] = `
<div
  className="app"
>
  <div
    className="days"
  >
    <div
      className="day"
    >
      <div
        className="dayname"
        onClick={[Function]}
      >
        Måndag
      </div>
      <div
        className="todo"
      >
        <div
          className="list"
        />
        <input
          onChange={[Function]}
          onKeyPressCapture={[Function]}
          value=""
        />
      </div>
    </div>
    <div
      className="day"
    >
      <div
        className="dayname"
        onClick={[Function]}
      >
        Tisdag
      </div>
      <div
        className="todo"
      >
        <div
          className="list"
        />
        <input
          onChange={[Function]}
          onKeyPressCapture={[Function]}
          value=""
        />
      </div>
    </div>
    <div
      className="day"
    >
      <div
        className="dayname"
        onClick={[Function]}
      >
        Onsdag
      </div>
      <div
        className="todo"
      >
        <div
          className="list"
        />
        <input
          onChange={[Function]}
          onKeyPressCapture={[Function]}
          value=""
        />
      </div>
    </div>
    <div
      className="day"
    >
      <div
        className="dayname"
        onClick={[Function]}
      >
        Torsdag
      </div>
      <div
        className="todo"
      >
        <div
          className="list"
        />
        <input
          onChange={[Function]}
          onKeyPressCapture={[Function]}
          value=""
        />
      </div>
    </div>
    <div
      className="day"
    >
      <div
        className="dayname"
        onClick={[Function]}
      >
        Fredag
      </div>
      <div
        className="todo"
      >
        <div
          className="list"
        />
        <input
          onChange={[Function]}
          onKeyPressCapture={[Function]}
          value=""
        />
      </div>
    </div>
    <div
      className="day"
    >
      <div
        className="dayname"
        onClick={[Function]}
      >
        Lördag
      </div>
      <div
        className="todo"
      >
        <div
          className="list"
        />
        <input
          onChange={[Function]}
          onKeyPressCapture={[Function]}
          value=""
        />
      </div>
    </div>
    <div
      className="day"
    >
      <div
        className="dayname"
        onClick={[Function]}
      >
        Söndag
      </div>
      <div
        className="todo"
      >
        <div
          className="list"
        />
        <input
          onChange={[Function]}
          onKeyPressCapture={[Function]}
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="currentweek"
  >
    <div
      className="weekbutton"
      onClick={[Function]}
    >
      ‹
    </div>
    <div
      className="weeknr"
    >
      Vecka 
      1
    </div>
    <div
      className="weekbutton"
      onClick={[Function]}
    >
      ›
    </div>
  </div>
</div>
`;

exports[`renders correctly lastweek 1`] = `
<div
  className="app"
>
  <div
    className="days"
  >
    <div
      className="day"
    >
      <div
        className="dayname"
        onClick={[Function]}
      >
        Måndag
      </div>
      <div
        className="todo"
      >
        <div
          className="list"
        />
        <input
          onChange={[Function]}
          onKeyPressCapture={[Function]}
          value=""
        />
      </div>
    </div>
    <div
      className="day"
    >
      <div
        className="dayname"
        onClick={[Function]}
      >
        Tisdag
      </div>
      <div
        className="todo"
      >
        <div
          className="list"
        />
        <input
          onChange={[Function]}
          onKeyPressCapture={[Function]}
          value=""
        />
      </div>
    </div>
    <div
      className="day"
    >
      <div
        className="dayname"
        onClick={[Function]}
      >
        Onsdag
      </div>
      <div
        className="todo"
      >
        <div
          className="list"
        />
        <input
          onChange={[Function]}
          onKeyPressCapture={[Function]}
          value=""
        />
      </div>
    </div>
    <div
      className="day"
    >
      <div
        className="dayname"
        onClick={[Function]}
      >
        Torsdag
      </div>
      <div
        className="todo"
      >
        <div
          className="list"
        />
        <input
          onChange={[Function]}
          onKeyPressCapture={[Function]}
          value=""
        />
      </div>
    </div>
    <div
      className="day"
    >
      <div
        className="dayname"
        onClick={[Function]}
      >
        Fredag
      </div>
      <div
        className="todo"
      >
        <div
          className="list"
        />
        <input
          onChange={[Function]}
          onKeyPressCapture={[Function]}
          value=""
        />
      </div>
    </div>
    <div
      className="day"
    >
      <div
        className="dayname"
        onClick={[Function]}
      >
        Lördag
      </div>
      <div
        className="todo"
      >
        <div
          className="list"
        />
        <input
          onChange={[Function]}
          onKeyPressCapture={[Function]}
          value=""
        />
      </div>
    </div>
    <div
      className="day"
    >
      <div
        className="dayname"
        onClick={[Function]}
      >
        Söndag
      </div>
      <div
        className="todo"
      >
        <div
          className="list"
        />
        <input
          onChange={[Function]}
          onKeyPressCapture={[Function]}
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="currentweek"
  >
    <div
      className="weekbutton"
      onClick={[Function]}
    >
      ‹
    </div>
    <div
      className="weeknr"
    >
      Vecka 
      52
    </div>
  </div>
</div>
`;
